File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # consider the min as the height 
        # width = right - left 
        left = 0
        right = len(height) - 1
        result = 0
        while left < right:
            width = right - left
            min_height = min(height[right], height[left])
            max_area = min_height * width
            if max_area > result:
                result = max_area
            elif height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return result                

File: 0011-container-with-most-water/0011-container-with-most-water.dart
class Solution {
  int maxArea(List<int> height) {
  
    int leftWall = 0;
    int rightWall = height.length - 1;
    int maxWater = 0;
  
    while(leftWall < rightWall) {
    
        int area = min(height[leftWall], height[rightWall]) * (rightWall - leftWall);
    
        maxWater = max(area, maxWater);
    
        if(height[leftWall] < height[rightWall]) {
            leftWall++;
        }
        else {
            rightWall--;
        }
    }
  return maxWater;
}  
}